### ISSUE TITLE
Gradle warm about "A configuration is resolved in a user-managed thread"

### ISSUE THREAD

**Issue by Kevinrob** [ben-manes/gradle-versions-plugin#309]
When I run the gradle task, I get a lot of warning about:
> The configuration :xxxxx:implementationCopy was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a thread not managed by Gradle or from a different project.  See https://docs.gradle.org/5.3.1/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints for more details. This behaviour has been deprecated and is scheduled to be removed in Gradle 6.0.

Do you know what is the cause of this? Is it hard to resolve it?

PS: ❤️ thank for this plugin

**Comment by ben-manes** [comment]
> A configuration is resolved in a user-managed thread (i.e., a thread not managed by Gradle).

The removal of our own threading was done in v21. We had tried since it seemed to be thread-safe and Gradle's resolution is quite slow, but it wasn't safe and didn't improve performance very much. The real answer would be parallel resolution, but that enhancement was only done for jar downloads and not metadata resolution so it doesn't help us here.

> A task from one project directly resolves a configuration in another project.

This is a very invasive change (#263). It requires sub-project tasks and an aggregator task to compose their results. We wrote this plugin for Gradle 1.x, so not a surprise things would eventually change. Unfortunately none of us are actively working on it so PRs are welcome, otherwise we try to fix issues when we find the free time. Long term I'd love it if Gradle built-in this plugin's features.
